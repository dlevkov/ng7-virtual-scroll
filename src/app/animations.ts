import {
  trigger,
  transition,
  style,
  animate,
} from '@angular/animations';


export const scrollIn = trigger('scrollAnimation', [
  transition(':enter', [
    style({
      transform: 'translateY(-200%) scale(1.5)',
      background: 'black',
      color: 'transparent',
    }),
    animate('1s 200ms ease-out'),
  ]),
  transition(':leave', [
    animate('0.5s 200ms ease-out', style({ transform: 'translateY(100%)' })),
  ]),
]);
