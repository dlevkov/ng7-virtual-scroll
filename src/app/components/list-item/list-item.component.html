<mat-card class="card-width" [@scrollAnimation]>
  <mat-card-header>
    <mat-card-title >{{ item.title }}</mat-card-title>
  </mat-card-header>
  <mat-card-content> {{ item.content }} </mat-card-content>
</mat-card>
<mat-divider></mat-divider>
