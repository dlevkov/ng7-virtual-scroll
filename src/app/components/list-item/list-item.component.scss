.card-width:hover {
  transform: scale(1.01);
  transition: all 0.1s ease-in-out;
}
.card-width {
  width: 100%;
}
