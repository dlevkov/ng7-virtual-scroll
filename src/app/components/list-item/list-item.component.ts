import {
  Component,
  OnInit,
  Input,
} from '@angular/core';
import * as fromAnimations from '../../animations';
@Component({
  selector: 'app-list-item',
  templateUrl: './list-item.component.html',
  styleUrls: ['./list-item.component.scss'],
  animations: [fromAnimations.scrollIn, fromAnimations.fadeInOut],
})
export class ListItemComponent implements OnInit {
  @Input() item: { title: string; content: string };
  constructor() {}

  ngOnInit() {}

}
