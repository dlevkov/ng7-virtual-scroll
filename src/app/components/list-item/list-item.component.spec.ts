import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ListItemComponent } from './list-item.component';
import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

describe('ListItemComponent', () => {
  let component: ListItemComponent;
  let fixture: ComponentFixture<ListItemComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ListItemComponent],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [BrowserAnimationsModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListItemComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  test('should render input', () => {
    component.item = { title: 'tit', content: 'con' };
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
