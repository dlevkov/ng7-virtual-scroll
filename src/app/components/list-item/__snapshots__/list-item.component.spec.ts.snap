// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent should render input 1`] = `
<div
  class="ng-tns-c1-0"
  id="root1"
  ng-version="7.0.3"
>
  <mat-card
    class="card-width ng-trigger ng-trigger-scrollAnimation ng-animating"
    style=""
  >
    <mat-card-header
      class="ng-tns-c1-0"
    >
      <mat-card-title
        class="ng-tns-c1-0"
      >
        tit
      </mat-card-title>
    </mat-card-header>
    <mat-card-content
      class="ng-tns-c1-0"
    >
       con 
    </mat-card-content>
  </mat-card>
  <mat-divider
    class="ng-tns-c1-0"
  />
</div>
`;
